irods_http_client/user_group_operations.py

```diff
@@ -0,0 +1,809 @@
+import requests
+import json
+
+class UsersGroups:
+    def __init__(self, url_base: str):
+        """
+        Initializes DataObjects with a base url. 
+        Token is set to None initially, and updated when setToken() is called in irodsClient.
+        """
+        self.url_base = url_base
+        self.token = None
+
+
+    def create_user(self, name: str, zone: str, user_type: str='rodsuser'):
+        """
+        Creates a new user. Requires rodsadmin or groupadmin privileges.
+        
+        Parameters
+        - name: The name of the user to be created.
+        - zone: The zone for the user to be created.
+        - user_type (optional): Can be rodsuser, groupadmin, or rodsadmin. Defaults to rodsuser.
+
+        Returns
+        - A dict containing the HTTP status code and iRODS response.
+        - The iRODS response is only valid if no error occurred during HTTP communication.
+        """
+        if (self.token == None):
+            raise RuntimeError('No token set. Use setToken() to set the auth token to be used')
+        if (not isinstance(name, str)):
+            raise TypeError('name must be a string')
+        if (not isinstance(zone, str)):
+            raise TypeError('zone must be a string')
+        if (not isinstance(user_type, str)):
+            raise TypeError('user_type must be a string')
+        if ((user_type != 'rodsuser') and (user_type != 'groupadmin') and (user_type != 'rodsadmin') and (user_type != '')):
+            raise ValueError('user_type must be set to rodsuser, groupadmin, or rodsadmin.')
+
+        headers = {
+            'Authorization': 'Bearer ' + self.token,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+
+        data = {
+            'op': 'create_user',
+            'name': name,
+            'zone': zone,
+            'user-type': user_type
+        }
+
+        r = requests.post(self.url_base + '/users-groups', headers=headers, data=data)
```

Comment:
A small optimization.

`self.url_base + '/users-groups'` can be defined as a class member and used throughout the implementation.

Open an issue for doing that for all endpoints. We'll handle it in a separate PR. Link the issue to this conversation.

Reply:
Opened issue #43